File: C++/D105.cpp
#include <iostream>
#include <list>
using namespace std;

int main() {
    list<int> d1(3);
    list<int> d2(3);
    int num;


    for (int i = 0; i < 3; i++) {
        cin >> num;
        d1.push_back(num);
    }

    for (int i = 0; i < 3; i++) {
        cin >> num;
        d2.push_back(num);
    }

    if (d1>d2)
        cout << "After" << endl;
    else if (d2>d1)
        cout << "Before" << endl;
    else
        cout << "Same" << endl;

    return 0;
}

File: Java/D105.java
import java.util.ArrayList;
import java.util.Scanner;

public class program {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        ArrayList<Integer> d1 = new ArrayList<>();
        ArrayList<Integer> d2 = new ArrayList<>();

        for (int i = 0; i < 3; i++) {
            d1.add(scanner.nextInt());
        }

        for (int i = 0; i < 3; i++) {
            d2.add(scanner.nextInt());
        }

        if (compareDates(d1, d2) > 0) {
            System.out.println("After");
        } else if (compareDates(d1, d2) < 0) {
            System.out.println("Before");
        } else {
            System.out.println("Same");
        }
    }

    public static int compareDates(ArrayList<Integer> d1, ArrayList<Integer> d2) {
        for (int i = 0; i < 3; i++) {
            if (d1.get(i) > d2.get(i)) {
                return 1;
            } else if (d1.get(i) < d2.get(i)) {
                return -1;
            }
        }
        return 0;
    }
}
